src/ray/object_manager/object_buffer_pool.h

```diff
@@ -0,0 +1,171 @@
+#ifndef RAY_OBJECT_MANAGER_OBJECT_BUFFER_POOL_H
+#define RAY_OBJECT_MANAGER_OBJECT_BUFFER_POOL_H
+
+#include <list>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include <boost/asio.hpp>
+#include <boost/asio/error.hpp>
+#include <boost/bind.hpp>
+
+#include "plasma/client.h"
+#include "plasma/events.h"
+#include "plasma/plasma.h"
+
+#include "ray/id.h"
+#include "ray/status.h"
+
+namespace ray {
+
+/// \class ObjectBufferPool Exposes chunks of object buffers for use by the ObjectManager.
+class ObjectBufferPool {
+ public:
+  /// Information needed about each object chunk.
+  /// This is the structure returned whenever an object chunk is
+  /// retrieved.
+  struct ChunkInfo {
+    ChunkInfo() {}
+    ChunkInfo(uint64_t chunk_index,
+              uint8_t *data,
+              uint64_t buffer_length)
+        : chunk_index(chunk_index),
+          data(data),
+          buffer_length(buffer_length){};
+    /// A pointer to the start position of this object chunk.
+    uint64_t chunk_index;
+    /// A pointer to the start position of this object chunk.
+    uint8_t *data;
+    /// The size of this object chunk.
+    uint64_t buffer_length;
+  };
+
+  /// Constructor.
+  ///
+  /// \param store_socket_name The socket name of the store to which plasma clients
+  /// connect.
+  /// \param chunk_size The chunk size into which objects are to be split.
+  ObjectBufferPool(const std::string &store_socket_name, const uint64_t chunk_size, const int release_delay);
+
+  ~ObjectBufferPool();
+
+  /// This object cannot be copied due to pool_mutex.
+  RAY_DISALLOW_COPY_AND_ASSIGN(ObjectBufferPool);
+
+  /// \param data_size The size of the object + metadata.
+  /// \return The number of chunks into which the object will be split.
+  uint64_t GetNumChunks(uint64_t data_size);
+
+  /// \param chunk_index The chunk index for which to obtain the buffer length.
+  /// \param data_size The size of the object + metadata.
+  /// \return The number of chunks into which the object will be split.
```

Comment:
Update this documentation. Also, generally it's nice to start with one line that describes the method's purpose, before the params.